cumulus/parachains/runtimes/assets/asset-hub-rococo/src/lib.rs

```diff
@@ -1799,6 +1799,18 @@ impl_runtime_apis! {
 	}
 }
 
+fn get_assets_in_pool_with(asset: xcm::v4::Location) -> impl Iterator<Item = AssetId> {
```

Comment:
I am not sure about using `xcm::v4::Location` and returning `xcm::latest::AssetId`. Now it is the same, latest == xcmV4, but what about upgrading just xcm to the xcmV5? So just the AssetId would be from v5? Now `asset_2.clone().into()` works just because of latest == xcmV4, but with xcmV5, you will need to do some conversion anyway. 

I think this function will be copied to all AssetHub runtimes, what about extracting it e.g. to the [`assets-common`](https://github.com/paritytech/polkadot-sdk/tree/master/cumulus/parachains/runtimes/assets/common) module and make it generic with `TryInto<Location>` or whatever feature. Or maybe add this filter function directly to the `pallet-asset-conversion`?

Reply:
Sadly I can't do that since the pools have an id of `T::PoolId`, it's only the asset hubs that give it the type of `(T::AssetKind, T::AssetKind)`.